src/cli/createDidConfig.ts

```diff
@@ -41,22 +51,39 @@ const createCredentialOpts = {
   wsAddress: { alias: 'w', type: 'string', demandOption: true, default: 'wss://spiritnet.kilt.io' },
 } as const
 
-async function issueCredential(did: DidUri, origin: string, seed: string, keyType: KeyType, nodeAddress: string) {
-  await connect(nodeAddress)
-  const didDocument = await Did.resolve(did)
-  const assertionMethod = didDocument?.document?.assertionMethod?.[0]
+async function issueCredential(did: string, origin: string, seed: string, keyType: KeyType, proofType?: string) {
+  const { didDocument } = await Did.resolveCompliant(did as DidUri)
+  const assertionMethodId = didDocument?.assertionMethod?.[0]
+  const assertionMethod = didDocument?.verificationMethod?.find(({ id }) => id.endsWith(assertionMethodId ?? '<none>'))
   if (!assertionMethod) {
     throw new Error(
       `Could not resolve assertionMethod of ${did}. Make sure the DID is registered to this chain and has an assertionMethod key.`
     )
   }
-  const keypair = new Keyring({ type: keyType }).addFromUri(seed)
-  if (assertionMethod.type !== keypair.type || !u8aEq(assertionMethod.publicKey, keypair.publicKey)) {
-    throw new Error('public key and/or key type of the DIDs assertionMethod does not match the supplied signing key')
+  // generate keypair and extract private key
+  const keypair = new Keyring().addFromUri(seed, undefined, keyType)
+  const { secretKey } = decodePair(undefined, keypair.encodePkcs8(), 'none')
+  if (!u8aEq(keypair.publicKey, base58Decode(assertionMethod.publicKeyBase58))) {
+    throw new Error('seed does not match DID assertion method')
   }
-  const keyUri: DidResourceUri = `${didDocument!.document!.uri}${assertionMethod.id}`
-  const signCallback: SignCallback = async ({ data }) => ({ signature: keypair.sign(data), keyUri, keyType })
-  const credential = await createCredential(signCallback, origin, did)
+  // create signer
+  const keyUri: DidResourceUri = assertionMethod.id
+  const createSigner = {
+    sr25519: sr25519Signer,
+    ed25519: eddsaSigner,
+    ecdsa: es256kSigner,
+  }
+  const signer = await createSigner[keyType]({ id: keyUri, secretKey, publicKey: keypair.publicKey })
+
+  const credential = await createCredential(
+    signer,
+    origin,
+    didDocument as ConformingDidDocument,
```

Comment:
Why has this been changed to a conforming document? Shouldn't the create credentials accept a did document?

Reply:
that's a Did document, the old SDK just has a weird name for it

Reply:
I took the decision here to build everything on W3C primitives instead of the slightly deviant implementation in the sdk

Reply:
Ok, that is fair enough.